docs/design/2021-10-20-table-attributes.md

```diff
@@ -0,0 +1,108 @@
+
+```sql
+ALTER TABLE t PARTITION p ATTRIBUTES[=]'key=value[, key1=value1...]';
+```
+
+And reset it by using:
+
+```sql
+ALTER TABLE t PARTITION p ATTRIBUTES[=]DEFAULT;
+```
+
+The attributes here can be any string that doesn't contain the reserved words. 
+
+There is no need for TiDB to know about the actual meaning of these attributes. The only thing that TiDB should do is to construct a request which contains the ID, the table/partition key range, the attributes, etc. Then synchronizing it with PD through the DDL job. All of these will be persisted in PD.
+
+
+
+### Display Attributes
+
+we add a new table in the `INFORMATION_SCHEMA` named `ATTRIBUTES`. If necessary, we can also support syntax like
+
+```sql
+SHOW ATTRIBUTES FOR [{DATABASE|SCHEMA} s][TABLE t][PARTITION p];
+```
+or 
+```
+SHOW CREATE TABLE;
+```
+
+At least we should support one way to show the attributes. Here is an example:
+
+```sql
+SELECT * FROM INFORMATION_SCHEMA.ATTRIBUTES;
++-------------------+-----------+----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| ID                | TYPE      | ATTRIBUTES           | RANGES                                                                                                                                                                                                                                   |
++-------------------+-----------+----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| schema/test/t1/p0 | key-range | "key=value"          | [7480000000000000ff3a5f720000000000fa, 7480000000000000ff3b5f720000000000fa]                                                                                                                                                             |
+| schema/test/t1    | key-range | "key=value"          | [7480000000000000ff395f720000000000fa, 7480000000000000ff3a5f720000000000fa], [7480000000000000ff3a5f720000000000fa, 7480000000000000ff3b5f720000000000fa], [7480000000000000ff3b5f720000000000fa, 7480000000000000ff3c5f720000000000fa] |
++-------------------+-----------+----------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+```
+
+### DDL Management
+Here are the concrete behaviors when doing the DDL operations with a table/partition having the attributes.
+
+#### Drop Clause 
++ For the drop table operation, the attributes will be deleted once the GC has finished
```

Comment:
```suggestion
+ For the drop schema and table operations, the attributes will be deleted once the GC has finished
```

Reply:
Actually, when dropping schema, the attributes will be deleted immediately.

Reply:
@rleungx well in that case update line 75 😓 

i think the attributes of the schema will be deleted immediately, but the attributes of the tables inside the schemas will need to wait for GC.

Reply:
updated